flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -100,6 +104,21 @@ private static Optional<String> firstPresent(Optional<String>... errOpts) {
         return Optional.empty();
     }
 
+    private Optional<String> validateDeploymentName(String name) {
+        if (name == null) {
+            return Optional.of("FlinkDeployment name must be define.");
```

Comment:
The null check is unnecessary, this will never be null on kubernetes

Reply:
Ok, I removed the null check.